qiskit_experiments/library/randomized_benchmarking/sampling_utils.py

```diff
@@ -0,0 +1,337 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Utilities for sampling layers in randomized benchmarking experiments
+"""
+
+import warnings
+import math
+from abc import ABC, abstractmethod
+from typing import Optional, Union, List, Tuple, Sequence, TypeVar
+from collections import defaultdict
+
+from numpy.random import Generator, default_rng, BitGenerator, SeedSequence
+import numpy as np
+
+from qiskit.circuit import Instruction
+from qiskit.exceptions import QiskitError
+
+from .clifford_utils import CliffordUtils, _CLIFF_SINGLE_GATE_MAP_1Q
+
+GateTypeT = TypeVar("GateTypeT", str, int, Instruction)
```

Comment:
`type` is missing? (`CXGate` seems to be allowed for `GateTypeT`)

Reply:
Oh, I was using the gate class itself instead of an instance because `rng.choice` interpreted the gate instances as arrays, but I fixed that by specifying the type in 09fe450. Now the input should be `CXGate()` instead of `CXGate`, and I check that the input gates are either "clifford", "pauli", or an `Instruction`.